openai/client.py

```diff
@@ -371,8 +371,153 @@ async def aimage_edit(
             response_format=response_format,
             user=user,
         )
-        return typing.cast(openai.Image, await openai.Image.acreate_edit(**kwargs))
+        return cast(openai.Image, await openai.Image.acreate_edit(**kwargs))
 
+    def edit(
+        self,
+        instruction: str,
+        *,
+        input: str = ...,
+        n: int = ...,
+        temperature: float = ...,
+        top_p: float = ...,
+        **kwargs,
+    ):
+        self._populate_args(
```

Comment:
It is a bit of an annoying pattern to have to do this dance. But any "magic" (e.g. decorator) would not be significantly simpler. It might be possible to inspect the decorated function and use annotations to avoid the duplication of signatures across, but that does feel pri3:ish.